trin-bridge/src/pandaops.rs

```diff
@@ -0,0 +1,135 @@
+use crate::bridge::Retry;
+use crate::full_header::{FullHeader, FullHeaderBatch};
+use anyhow::{anyhow, bail};
+use ethereum_types::H256;
+use ethportal_api::types::jsonrpc::params::Params;
+use ethportal_api::types::jsonrpc::request::JsonRequest;
+use ethportal_api::utils::bytes::hex_encode;
+use ethportal_api::{Header, Receipts};
+use serde_json::{json, Value};
+
+pub struct PandaOpsMiddleware {
+    pub base_el_endpoint: String,
+    pub base_cl_endpoint: String,
+    pub client_id: String,
+    pub client_secret: String,
+}
+
+impl PandaOpsMiddleware {
+    pub fn new(
+        base_el_endpoint: String,
+        base_cl_endpoint: String,
+        client_id: String,
+        client_secret: String,
+    ) -> Self {
+        Self {
+            base_el_endpoint,
+            base_cl_endpoint,
+            client_id,
+            client_secret,
+        }
+    }
+}
+
+impl PandaOpsMiddleware {
+    /// Used the "surf" library here instead of "ureq" since "surf" is much more capable of handling
+    /// multiple async requests. Using "ureq" consistently resulted in errors as soon as the number of
+    /// concurrent tasks increased significantly.
+    async fn batch_request(&self, obj: Vec<JsonRequest>) -> anyhow::Result<String> {
```

Comment:
I guess, if the middleware now has this concept of el & cl providers, we might want to change the names of `batch_request` / `request` to indicate that they are tied to a specific kind of provider. Maybe `el_batch_request` / `cl_request`? Idk, there's probably also a further refactoring opportunity here where we have an actual `Provider` trait that implements `request` & `batch_request` for both el & cl providers... but I'm not convinced that's something we need to implement at this point.

Reply:
Yeah, I'm planning to refactor this further when we implement the consensus data retrieval.